Intro table updates 4

seanforyou23 opened the pull request:
This PR applies some updates to the intro-table tutorial in response to recent feedback we've received. It's mostly cleaning up step instruction, but also applies a global change in the way we refer to input properties for react components. Previously we used the short version of "prop" and "props", now we use the full word properties which are a little less likely to be misunderstood.

Should help close https://github.com/patternfly/training-scenarios/issues/270, https://github.com/patternfly/training-scenarios/issues/254 and https://github.com/patternfly/training-scenarios/issues/171. 

https://www.katacoda.com/seanforyou23/courses/react-components/table-intro

Lemme know if you see any issues!

christiemolloy commented on react-components/table-intro/step1.md:
```diff
@@ -7,7 +7,7 @@ The following is an example of what the table component structure looks like:
 </pre>
 
-Columns can be expressed as an array of strings, or an array of objects which match the type "ICell". The columns definition is passed to the `cells` prop of the Table component which can take the form of `Array<ICell | string>`. Below is what an ICell looks like.
+Columns can be expressed as an array of strings or an array of objects which match the type "ICell". An "ICell" is a TypeScript interface that represents a cell of the `<Table>`. The columns definition is passed to the `cells` property of the `<Table>` component which can take the form of `Array<ICell | string>`. Below is what the ICell interface looks like.
```
`An "ICell is a TS interface......` still feels a little like a concept outside of PatternFly, how about: "ICell" is the name given to the Typescript interface that represents a cell of the `<Table>` . Otherwise reads much better.

I wonder if ICell should be wrapped in back-ticks or in quotation marks here? It is referring to code, but also to the name, but typically i think we have gone with wrapping it as code.

christiemolloy commented on react-components/table-intro/step1.md:
```diff
@@ -45,7 +45,7 @@ const columnsDefinition = [
 </pre>
 
-Rows can be expressed as a multidimensional array of strings, or as an array of objects which match the type "IRow". The rows definition is passed to the `rows` prop of the Table or TableBody component, and can take the form of `Array<IRow | string[]>`. Below is what an IRow should look like.
+Rows can be expressed as a multidimensional array of strings or as an array of objects which match the type "IRow". An "IRow" is a TypeScript interface that represents a row of the `<Table>`. The rows definition is passed to the `rows` property of the `<Table>` or `<TableBody>` component, and can take the form of `Array<IRow | string[]>`. Below is what an IRow should look like.
```
similar comment here for iRow, as above

christiemolloy commented on react-components/table-intro/step2.md:
```diff
@@ -49,3 +49,5 @@ cells={columns} rows={defaultRows}
 
 <img src="table-intro/assets/step-2-complete.png" alt="Image of what table looks like at the end of step 2." style="box-shadow: rgba(3, 3, 3, 0.2) 0px 1.25px 2.5px 0px;" />
+
+Still have questions? View the latest <a href="https://www.patternfly.org/v4/documentation/react/components/table/" target="_blank">Table documentation</a> for more information.
```
Should this be []() syntax for links in markdown?

seanforyou23 replied:
Oh geez, good catch! Updated to markdown syntax

christiemolloy commented on react-components/table-intro/step4.md:
```diff
@@ -38,15 +38,15 @@ const defaultRows = [
 For each array item in each row's `cells` array, replace the string value with an object whose `title` property is the row cell value.
 
-It should look like this:
+Each array item in each row’s cells array looks like this:
```
might help with the awkwardness of the sentence

```suggestion
For every array item in each row’s cells array looks like this:
```

seanforyou23 replied:
Revised this section a bit for clarity, let me know what you think

christiemolloy commented on react-components/table-intro/step4.md:
```diff
@@ -38,15 +38,15 @@ const defaultRows = [
 </pre>
 
-becomes
+After replacing the string value with an object whose title property is the row cell value, the code should look like this:
```
```suggestion
After replacing the string value with an object that has a title property equal to the row cell value, the code should look like this:
```

christiemolloy commented on react-components/table-intro/step5.md:
```diff
@@ -1,8 +1,8 @@
-A typical case for the table component is the need to handle large datasets elegantly. This is addressed on both the styling and interactivity side. In this step, styling techniques are applied to improve the experience.
+Table components are commonly used to handle large datasets. This is addressed on both the styling and interactivity side. In this step, apply styling techniques to improve the experience.
```
I think styling techniques is a little broad here

seanforyou23 replied:
I revised this section also, think it's clearer now. Let me know what you think!

christiemolloy commented on react-components/table-intro/step8.md:
```diff
@@ -1,8 +1,8 @@
-This step adds the state properties needed to track how many rows per page the user can display and what current page the user is on within the dataset. It also introduces a function to handle updating the number of rows per page to show when the user changes this value with the dropdown.
+In this step, add state properties to track how many rows per page the user can display and what current page the user is on within the dataset. Also, introduce a function to handle updating the number of rows per page to show when the user changes this value with the dropdown. IThis tutorial uses the React <a href="https://reactjs.org/docs/hooks-overview.html" target="_blank">hooks api</a>, howe ...
```
Spelling mistake: IThis

seanforyou23 commented on react-components/table-intro/step3.md:
```diff
@@ -13,3 +13,5 @@ Apply this transformation to all of the column definitions.
 <strong>Note: </strong> The table should appear exactly as it did before:
```
I added some more information to this step to make it more obvious what the user is doing and why. Hope that helps, let me know what you think.

We separated the transform of column format and row format into multiple steps because transforming the rows is a bit more lengthy to explain, figured by having people do the exercise with the simpler columns first, the next step wouldn't be such a shock. WDYT?

dlabrecq replied:
I didn't notice the "Show solution" button at the bottom, but that would be fine.

seanforyou23 commented:
Thx for the feedback! I just sent some more updates, ready for another review when there is time. 

@dlabrecq I'm with you on dialing in the right balance of helpful interaction controls. We've added and removed them from a few places a few times, think there were some issues with line breaks and the "copy to clipboard" text covering parts of the code snippets when the snippets are only a few lines long. I tried to limit the use of copy to editor to the beginning on the tutorial to avoid people overwriting the whole tutorial by accident (or using this as a shortcut). I can add back those links to auto-open files, I'll have a look through the history first though to remember why we removed them in the first place.

dlabrecq commented:
Whatever the team wants regarding interaction controls is fine, I just wanted to note that this is a different experience than other tutorials.
